server/src/api/customers.ts

```diff
@@ -58,7 +59,17 @@ customerRouter.post(
   '/',
   customerAuth,
   async (req: Request, res: Response, next: NextFunction) => {
-    const customerData: CustomerPayload = req.body;
+    // TODO: Ensure that req.body is of type CustomerPayload
+    req.validated = req;
+    next();
+  },
+  async (req: Request, res: Response, next: NextFunction) => {
+    req.authorizedCustomerGPayId = req.validated.body.gpayId;
```

Comment:
looks like req.validated.body.gpayId is also from spot api identity?
if so, then customerAccessControl is not needed here since inside it's comparing two ids from the same source?

Reply:
@ifhuang `req.validated.body.gpayId` is the `gpayId` field in the request body. From the client Group Buy app, it is indeed retrieved from api identity, but on the server side, we cannot just trust the input from the request, because another user can just replay the request with their own token, but with my gpayId.